src/librustc_typeck/coherence/mod.rs

```diff
@@ -258,9 +258,8 @@ impl<'a, 'gcx, 'tcx> CoherenceChecker<'a, 'gcx, 'tcx> {
                                                    type `{}`",
                                                    self_type);
                     err.span_label(span,
-                                   &format!("field `{}` in struct `{}` doesn't implement `Copy`",
-                                            name,
-                                            self_type));
+                                   &format!("field `{}` doesn't implement `Copy`",
+                                            name));
                     if let Some(def_span) = tcx.map.span_if_local(did) {
                         err.span_label(def_span, &"this field doesn't implement `Copy`");
                     }
```

Comment:
I was thinking something more like:

```rust
if let Some(def_span) = tcx.map.span_if_local(did) {
    err.span_label(def_span, &"this field doesn't implement `Copy`");
    err.span_label(span, &format!("field `{}` doesn't implement `Copy`", name));
} else {
    err.span_label(span,
                   &format!("field `{}` in struct `{}` doesn't implement `Copy`",
                            name,
                            self_type));
}
```

That way, in the case we don't have the def_span available, we can still give them a little more hint as to where it's coming from.